### PR TITLE
Tokens that fail to redeem due to network errors should be added to a queue and retried

### PR DESCRIPTION
**Pull request by tmancey**
fixes https://github.com/brave/brave-browser/issues/3662

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests && npm run test-security`) on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone

## Test Plan:

Confirm failed requests are added to the `Defaults/rewards_service/confirmations.json` under `confirmations`/`failed_confirmations` and confirmation redemption is retried every 5 minutes +/- jitter until successful. Items that are retried and fail should move to the back of the queue.

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

### CODE REVIEW

File: vendor/bat-native-ads/include/bat/ads/confirmation_type.h [hunk 1]
```diff
@@ -6,19 +6,34 @@
...
+  enum Value : int {
+    UNKNOWN,
+    CLICK,
+    DISMISS,
+    VIEW,
+    LANDED
+  };
+
+  ConfirmationType() = default;
+  constexpr ConfirmationType(const Value value) : value_(value) {}
```
Comment (yrliou) [inline vendor/bat-native-ads/include/bat/ads/confirmation_type.h]: lint error here, Single-parameter constructors should be marked explicit.  [runtime/explicit] [5]

File: vendor/bat-native-confirmations/include/bat/confirmations/confirmation_type.h [hunk 2]
```diff
@@ -6,19 +6,34 @@
...
+  enum Value : int {
+    UNKNOWN,
+    CLICK,
+    DISMISS,
+    VIEW,
+    LANDED
+  };
+
+  ConfirmationType() = default;
+  constexpr ConfirmationType(const Value value) : value_(value) {}
```
Comment (yrliou) [inline vendor/bat-native-confirmations/include/bat/confirmations/confirmation_type.h]: DITTO.

File: vendor/bat-native-ads/include/bat/ads/confirmation_type.h (hunk 1 above)
Comment (yrliou) [reply vendor/bat-native-ads/include/bat/ads/confirmation_type.h]: as we discussed in slack, I'm fine with NOLINT and a comment stating that we allow implicit conversion from enum value to this wrapper class, similar as https://cs.chromium.org/chromium/src/v8/src/wasm/value-type.h?rcl=88a0c6d607cf023a5776193fe1786be0ca9c0767&l=64-65.

Comment (tmancey) [reply vendor/bat-native-ads/include/bat/ads/confirmation_type.h]: 👍 added NOLINT and comment as above, thanks

File: vendor/bat-native-confirmations/include/bat/confirmations/confirmation_type.h (hunk 2 above)
Comment (tmancey) [reply vendor/bat-native-confirmations/include/bat/confirmations/confirmation_type.h]: 👍 added NOLINT and comment as above, thanks

**Pull request merged**